src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -507,35 +517,36 @@ void func_80B0D3AC(EnSw* this, PlayState* play) {
 
     Math_ApproachF(&this->actor.scale.x, 0.02f, 0.2f, 0.01f);
     Actor_SetScale(&this->actor, this->actor.scale.x);
-    this->actor.world.pos.x += this->unk_364.x * this->actor.velocity.y;
-    this->actor.world.pos.y += this->unk_364.y * this->actor.velocity.y;
-    this->actor.world.pos.z += this->unk_364.z * this->actor.velocity.y;
+    this->actor.world.pos.x += this->wallPolyNormal.x * this->actor.velocity.y;
+    this->actor.world.pos.y += this->wallPolyNormal.y * this->actor.velocity.y;
+    this->actor.world.pos.z += this->wallPolyNormal.z * this->actor.velocity.y;
     this->actor.world.pos.x += this->unk_37C.x * this->actor.speedXZ;
     this->actor.world.pos.y += this->unk_37C.y * this->actor.speedXZ;
     this->actor.world.pos.z += this->unk_37C.z * this->actor.speedXZ;
     this->actor.velocity.y += this->actor.gravity;
     this->actor.velocity.y = CLAMP_MIN(this->actor.velocity.y, this->actor.minVelocityY);
 
     if (this->actor.velocity.y < 0.0f) {
-        this->unk_360 = 0;
+        this->goldHiddenBool = 0;
```

Comment:
```suggestion
        this->goldHiddenBool = false;
```
Same everywhere else it is used.